feature-list-streams/src/main/java/com/codandotv/streamplayerapp/feature_list_streams/search/presentation/screens/SearchViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package com.codandotv.streamplayerapp.feature_list_streams.search.presentation.screens
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.codandotv.streamplayerapp.core_networking.handleError.catchFailure
+import com.codandotv.streamplayerapp.feature_list_streams.list.domain.GetTopRatedStream
+import com.codandotv.streamplayerapp.feature_list_streams.search.domain.usecase.SearchUseCase
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.launch
+
+class SearchViewModel(
+    private val searchUseCase: SearchUseCase,
+    private val latestStream: GetTopRatedStream
+): ViewModel(), DefaultLifecycleObserver {
+
+//    private val _isShowstateSearch: MutableState<SearchUIState> = mutableStateOf(false)
+//    val isShowstateSearch: MutableState<Boolean> = _isShowstateSearch
+
+    private val _currentSearchText = MutableStateFlow("")
+    val currentSearchText = _currentSearchText.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            /*
+            Substituir esse chamada ao endpoint de pesquisa pelo o de principais buscas,
+            Devemos substituir o parametro query com o texto que usuário está digitando
+             */
+            searchUseCase(query = "i").onStart {
```

Comment:
**#question**
- Is this PR a work in progress? If yes we can add a prefix `WIP`. When it is available for review, you can remove the `WIP`.

Reply:
esse i seria o q? c
além disso, eu acho que não estamos usando o init {}

tem  o ciclo de vida do viewmodel, acho que eu fiz no detail da uma olhada

Reply:
@gabrielbmoro valeu pelo toque, adicionei ali
@rodrigoliveirac o endpoint não aceitava uma busca com valor vazio, então tinha chumbado (letra "i") enquanto estava sendo desenvolvido a feature , porém já foi removido e ajustado o cenário.

Reply:
eu sou o rviannaoliveira hahaha